### PR TITLE
Architecture notes: Networking

### PR DESCRIPTION
**Pull request by ctarda**
Implements part of #287

This PR contains an architectural overview of Networking.

If and when it is considered worth incorporating into the project, we will do it to the wiki directly.

### CODE REVIEW

File: docs/NETWORKING.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+![Networking high level class diagram](images/networking.png)
+
+## Remote
+A `Remote` performs network requests. This is the core of this module.  
+
+The base implementation of `Remote`  is provided an implementation of the `Network` protocol, and enqueues pairs of `Request` and `Mapper`, executing that request on the provided `Network` and delegating the parsing of the response to the `Mapper` provided. 
+
+There is a subclass of `Remote` for each relevant concern. Each of these subclasses provide public methods for each of the networking operations relevant to that concern.
+
+These are the current subclass of `Remote`:
```
Comment (bummytime) [inline docs/NETWORKING.md]: I would consider putting a qualifier on this list mentioning that there are probably more `Remotes`. Something like "at the time of writing this document". Just thinking of the likelihood of us making sure these docs are up-to-date. 😄

File: docs/NETWORKING.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+## Mapper
+A protocol that abstracts the different parsers.
+
+There are several implementations of this protocol, roughly one per Remote, although in some cases there is more than one implementation per remote (roughly, one for a single model object, and another for a collection of the same model object). 
+
+Mappers receive an instance of `Data` and return  the result of parsing that data as a model object or a collection of model objects.
+
+## Model objects
+Model objects declares in `Networking` are immutable, and modelled as value types (structs) that implement `Decodable`
```
Comment (bummytime) [inline docs/NETWORKING.md]: Maybe we should say:

> Model objects declared in `Networking` are immutable, and modeled as value types (structs) that _typically_ implement `Decodable`.

😄 

There are some circumstances where the models don't implement `Decodable`, e.g. `ShipmentTrackingProvider` and `ShipmentTrackingProviderGroup`

File: docs/NETWORKING.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+* `ShipmentsRemote` All things Shipment Tracking, from tracking providers to actual tracking associated to an order
+* `SiteAPIRemote` Loads the API information associated to a site.
+* `SiteSettingsRemote` Loads a site’s settings
+* `SiteVisitStatsremote` fetches visitor stats for a given site
+* `TopEarnersStatsRemote`fetches the top earner stats for a given site.
+
+## Network
+A protocol that abstracts the networking stack. 
+
+There are two implementations of this protocol: `AlamofireNetwork` which manages a networking stack based on the third party library Alamofire, and `MockupNetwork` which is a mock networking stack that does not actually hit the network, to be used in the unit tests.
```
Comment (bummytime) [inline docs/NETWORKING.md]: Maybe link to the AlamoFire project on GitHub?

**Comment by ctarda** [comment]
Thanks for the feedback @bummytime ! I did incorporate all your suggestions, and I will proceed to add this doc to the wiki

**Pull request closed without merging**